### PR TITLE
429 additional report metrics

### PR DESCRIPTION
**Pull request by js-xc**
Not sure what will happen if input reports are not using the same Pxx (usually 50/95/99/99.9), I will check this.

### CODE REVIEW

**Comment by jowerner** [comment]
Now that we have the Percentiles column set in the table, we can remove the Median column in the Runtime column set.

**Comment by jowerner** [comment]
In case there is no data available to show in a table, the then empty table misses some cells in the table body and footer.

**Comment by jowerner** [comment]
Add an asterisk to the 1/h and 1/d column headers in the Bandwidth table to indicate these values are projected.

**Comment by js-xc** [comment]
> Add an asterisk to the 1/h and 1/d column headers in the Bandwidth table to indicate these values are projected.

This applies to diffreport as well as loadtest report, right?

**Comment by jowerner** [comment]
> > Add an asterisk to the 1/h and 1/d column headers in the Bandwidth table to indicate these values are projected.
> 
> This applies to diffreport as well as loadtest report, right?

Right.

**Comment by jowerner** [comment]
While we are at it, could you please add an explanation for the two asterisks at the Distinct** column to the help text of the Requests page? Have a look at the load test report and borrow the text from there.

**Comment by jowerner** [comment]
In `config/xsl/loadreport/util/timer-row.xsl`, we show the "Distinct" column for requests only. Please simplify the template by using a plain `xsl:if` instead of `xsl:choose`.

**Comment by js-xc** [comment]
> In case there is no data available to show in a table, the then empty table misses some cells in the table body and footer.

I don't understand, what's wrong with this? Or is the html wrong and Firefox is just filling the gaps itself so that it looks ok?
Screenshot from 2024-01-15 13-50-10

**Comment by jowerner** [comment]
> > In case there is no data available to show in a table, the then empty table misses some cells in the table body and footer.
> 
> I don't understand, what's wrong with this? Or is the html wrong and Firefox is just filling the gaps itself so that it looks ok? Screenshot from 2024-01-15 13-50-10

See below:

Screenshot 2024-01-15 at 13-53-58 XLT Performance Comparison Report - Page Load Timings

File: config/xsl/diffreport/util/timer-table.xsl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	                                <xsl:value-of select="12 + $percentileCount"/>
+	                            </xsl:when>
+	                            <xsl:when test="$type = 'action'">
+	                                <xsl:value-of select="12 + $percentileCount"/>
+	                            </xsl:when>
+	                            <xsl:otherwise>
+	                                <xsl:value-of select="11 + $percentileCount"/>
+	                            </xsl:otherwise>
+	                        </xsl:choose>
+                        </xsl:variable>
```
Comment (jowerner) [inline config/xsl/diffreport/util/timer-table.xsl:97]: The calculated columns value looks incorrect to me, except for requests. But see the other comment below as well.

File: config/xsl/diffreport/util/timer-table.xsl [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                        <tfoot>
+                            <tr>
+                                <td colspan="{$columns}"></td>
+                            </tr>
+                            <xsl:call-template name="filtered-footer-row" />
+                        </tfoot>
```
Comment (jowerner) [inline config/xsl/diffreport/util/timer-table.xsl:103]: When using a single td with colspan, the column (set) striping effect as introduced in the table header is lost. We should have at least one td per column (set). Make sure to match the look with the Bandwidth table.

(I know that column striping in general is a mess in the diff report, but definitely another topic. Just wanted to make sure that we don't add to it.)

**Review by jowerner — APPROVED** [review]

**Pull request merged**